src/Microsoft.AspNet.Mvc.Core/DefaultControllerActionArgumentBinder.cs

```diff
@@ -98,24 +100,74 @@ public class DefaultControllerActionArgumentBinder : IControllerActionArgumentBi
             return modelBindingResult;
         }
 
+        // Called via reflection.
+        private static void CallPropertyAddRange<TElement>(object target, object source)
+        {
+            var targetCollection = (ICollection<TElement>)target;
+            var sourceCollection = source as IEnumerable<TElement>;
+            if (sourceCollection != null && !targetCollection.IsReadOnly)
+            {
+                targetCollection.Clear();
+                foreach (var item in sourceCollection)
+                {
+                    targetCollection.Add(item);
+                }
+            }
+        }
+
         private void ActivateProperties(object controller, Type containerType, Dictionary<string, object> properties)
         {
-            var propertyHelpers = PropertyHelper.GetProperties(controller);
+            var containerExplorer = _modelMetadataProvider.GetModelExplorerForType(containerType, controller);
             foreach (var property in properties)
             {
-                var propertyHelper = propertyHelpers.First(helper =>
-                    string.Equals(helper.Name, property.Key, StringComparison.Ordinal));
-                if (propertyHelper.Property == null || !propertyHelper.Property.CanWrite)
+                var propertyExplorer = containerExplorer.GetExplorerForProperty(property.Key);
+                var propertyMetadata = propertyExplorer?.Metadata;
+                if (propertyMetadata == null)
+                {
+                    // Nothing to do if property is unknown to the ModelMetadataProvider.
```

Comment:
removed while going back to `PropertyHelper` because the `PropertyHelper` constructor has `[NotNull]` for the `PropertyInfo`